packages/amesos2/src/Amesos2_TpetraMultiVecAdapter_def.hpp

```diff
@@ -208,10 +202,10 @@ namespace Amesos2 {
           const size_t lclNumRows = redist_mv.getLocalLength();
           for (size_t j = 0; j < redist_mv.getNumVectors(); ++j) {
             auto av_j = av(lda*j, lclNumRows);
-            auto X_lcl_j_2d = redist_mv.getLocalViewHost();
+            auto X_lcl_j_2d = redist_mv.getLocalViewHost(Tpetra::Access::ReadOnly);
             auto X_lcl_j_1d = Kokkos::subview (X_lcl_j_2d, Kokkos::ALL (), j);
 
-            using val_type = typename decltype( X_lcl_j_1d )::value_type;
+            using val_type = typename std::remove_const<typename decltype( X_lcl_j_1d )::value_type>::type;
```

Comment:
Could you use ``typename multivec_t::impl_scalar_type`` instead for readability?